models/accessom3.py

```diff
@@ -0,0 +1,85 @@
+"""Specific Access-OM3 Model setup and post-processing"""
+
+import hashlib
+from pathlib import Path
+from payu.models.cesm_cmeps import Runconfig
+from typing import Dict, Any
+
+from models.model import Model
+
+BASE_SCHEMA_URL = "https://raw.githubusercontent.com/ACCESS-NRI/schema/main/au.org.access-nri/model/access-om2/experiment/reproducibility/checksums"
+
+SCHEMA_VERSION_1_0_0 = "1-0-0"
+DEFAULT_SCHEMA_VERSION = SCHEMA_VERSION_1_0_0
+SUPPORTED_SCHEMA_VERSIONS = [SCHEMA_VERSION_1_0_0]
+
+class AccessOm3(Model):
+    def __init__(self, experiment):
+        super(AccessOm3, self).__init__(experiment)
+        self.output_file = self.experiment.output000 / 'ocean.stats'
+
+        self.runconfig = experiment.control_path / 'nuopc.runconfig'
+        self.ocean_config = experiment.control_path / 'input.nml'
+        self.default_schema_version = DEFAULT_SCHEMA_VERSION
+
+    def set_model_runtime(self,
+                          years: int = 0,
+                          months: int = 0,
+                          seconds: int = 10800):
+        """Set config files to a short time period for experiment run.
+        Default is 3 hours"""
+        runconfig = Runconfig(self.runconfig)
+
+        if years == months == 0:
+            freq = "nseconds"
+            n = str(seconds)
+        elif seconds == 0:
+            freq = "nmonths"
+            n = str(12 * years + months)
+        else:
+            raise NotImplementedError(
+                f"Cannot specify runtime in seconds and year/months at the same time")
+
+        runconfig.set("CLOCK_attributes", "restart_n", n)
+        runconfig.set("CLOCK_attributes", "restart_option", freq)
+        runconfig.set("CLOCK_attributes", "stop_n", n)
+        runconfig.set("CLOCK_attributes", "stop_option", freq)
+
+        runconfig.write()
+
+    def output_exists(self) -> bool:
+        """Check for existing output file"""
+        return self.output_file.exists()
+
+    def extract_checksums(self,
+                          output_directory: Path = None,
+                          schema_version: str = None) -> Dict[str, Any]:
+        """Parse output file and create checksum using defined schema"""
+        if output_directory:
+            output_filename = output_directory / 'ocean.stats'
+        else:
+            output_filename = self.output_file
+
+        # ocean.stats is used for regression testing in MOM6's own test suite
+        # See https://github.com/mom-ocean/MOM6/blob/2ab885eddfc47fc0c8c0bae46bc61531104428d5/.testing/Makefile#L495-L501
+        # Here we calculate the md5 hash of ocean.stats
+        with open(output_filename, 'rb') as f:
+            contents = f.read()
+            md5_hash = hashlib.md5(contents).hexdigest()
```

Comment:
Is there any chance checksumming the whole file will result in false positives/negatives, if say formatting changes?

Reply:
Also, is there any value in granularity in the tests? So knowing if particular tests failed? Or if they have changed, but only a little bit?

Reply:
> Is there any chance checksumming the whole file will result in false positives/negatives, if say formatting changes?

Yes, in this case the reference checksums would need to be regenerated I suppose. I'm not immediately sure what we could do instead. Extracting and checking contents would presumably also given bad behaviour if the formatting changes. Note, I chose to use the `ocean.stats` file here because that's what's used in MOM6's own regression testing, but there may be better outputs to check.

Reply:
> Also, is there any value in granularity in the tests? So knowing if particular tests failed? Or if they have changed, but only a little bit?

Maybe in the long run? I would also like to include output from other components, so this is definitely just a basic starting point.

Regarding small changes, I think any change means the model is not reproducing so should fail. But in the future we could include messaging about "how much" it's not reproducing.

Reply:
> Yes, in this case the reference checksums would need to be regenerated I suppose. I'm not immediately sure what we could do instead. Extracting and checking contents would presumably also given bad behaviour if the formatting changes. Note, I chose to use the `ocean.stats` file here because that's what's used in MOM6's own regression testing, but there may be better outputs to check.

Having looked at the `ocean.stats` file I'd say it would be a little more robust to extract each of the fields and save them individually. Could use the header strings as keys.

This would have the advantage of being invariant/insensitive to *some* formatting changes. If you removed the strings from the values in the third line and read them in as floats it would also make the checksum part invariant to changes in whitespace, but potentially not to changes in the formatting of the values themselves.

In the repro CI we're using the state of the checksum reproducibility to determine when to change major version: when the model is no longer reproducing. It would be bad to get false positives and change the major version when the model was still reproducing previous answers.

Reply:
Done in [0313093](https://github.com/ACCESS-NRI/model-config-tests/pull/19/commits/0313093d459eb9a3b91e3f6fffa52b19602f47d3). In doing this it became clear that something like this is actually needed for the restart repro tests to work so thanks for the suggestion :)